File: ReactNativeApp/android/app/src/main/java/com/components/basic/timer/TimerBuilder.java
package com.components.basic.timer;

import com.ssvc.SSVC;
import com.ssvc.SSVCBuilder;

public class TimerBuilder extends SSVCBuilder {

    public static SSVC<TimerState, TimerView, TimerController> buildTimer() {
        TimerState state = new TimerState();
        TimerView timerView = new TimerView(state);
        TimerController timerController = new TimerController(state);

        return new SSVC<>(state, timerView, timerController);
    }


}

File: ReactNativeApp/ios/Modules/SSVC/Instance/Timer/TimerBuilder.swift
//
//  TimerBuilder.swift
//  jogo
//
//  Created by Mohsin on 27/10/2021.
//

import Foundation

public class TimerBuilder: SSVCBuilder {
  
  public static func build() -> SSVC<TimerState, TimerView, TimerController> {
    let state = TimerState()
    let view = TimerView(state: state)
    let controller = TimerController(state: state)
    return SSVC(state: state, stateView: view, controller: controller)
  }
}
